code/modules/cargo/reward.dm

```diff
@@ -0,0 +1,75 @@
+/datum/bounty_reward/
+	var/name = "Reward"
+	var/id
+	var/reward_description = "loot" //The description of the reward
+	var/crate_access = list() //Required access to open the crate, if any.
+	var/obj/structure/closet/crate/spawn_crate //The crate to spawn in.
+	var/list/item_rewards_flat = list() //Always give these rewards
+	var/list/item_rewards_weight = list() //Pick from a list of these rewards based on weights.
+	var/reward_count = 0 //Number of items to pick from item rewards weight
+
+/datum/bounty_reward/proc/spawnReward(var/spawn_loc,var/high_priority)
+
+	var/reward_mod = high_priority ? 2 : 1
+
+	if(spawn_crate)
+		var/obj/structure/closet/crate/spawned_crate = new spawn_crate(spawn_loc)
+		spawn_loc = spawned_crate
+		spawn_crate.req_access = crate_access
+		spawn_crate.req_one_access = list()
+
+	if(item_rewards_flat.len)
+		for(var/spawned_item in item_rewards_flat)
+			new spawned_item(spawn_loc)
+
+	if(reward_count && item_rewards_weight.len)
+		for(var/i=1,i <= reward_count*reward_mod,i++)
+			var/chosen_item = pickweight(item_rewards_weight)
+			new chosen_item(spawn_loc)
+
+/datum/controller/subsystem/cargo/proc/setupRewards()
+	for(var/datum/bounty_reward/reward in subtypesof(/datum/bounty_reward/))
+		if(!reward.id)
+			continue
+		rewards_list[reward.id] = reward
+
+/datum/controller/subsystem/cargo/proc/spawnReward(var/reward_id)
```

Comment:
That should be reworked to ammend the shipment manifest with the items that have been placed on the shuttle as bounty.

Possibly generate a message here `selected_reward.spawnReward(pick(clear_turfs))` and then just ammend it to to the manifest.